### DISCUSSION TITLE
Multi GPU plotting with weird GPU order

### DISCUSSION THREAD

**Discussion by JuseyMiners** [madMAx43v3r/chia-gigahorse#199] [Q&A]
Hey all, 

I have a server with LOTS of ram and 5 GPUs. It has a 3070 Ti and 4x tesla P4 gpus.

I would like to have two plotters running; one that uses the 3070Ti and another that gangs together the p4s to work together. The way I understand it though, all of the collaborated GPUs must be sequential from the OS perspective. 

Unfirtunately, I have 2 p4, then 1 3070ti then 2 more p4 from ubuntu's perspective. NVtop shows gpu 2 as 3070ti and gpus 0,1,3,4 as tesla p4 gpus. 

I can't move the GPUs around because the 3070Ti can only fit in the slot that it is in, and all other slots are populated with p4s. 

Is there a way to gang the 4 P4s together even though they are not sequential? Or do I have to run 3 plot operations and separate them out?

**Comment by Kruger1981** [comment]
To use the 3070 use -g 2

Multi-GPU
To plot with multiple GPUs the -r flag is used, for example to plot with the first two GPUs:

cuda_plot_kxx -r 2 ...
Or to plot with GPU 1 and 2 (ie. second and third)

cuda_plot_kxx -g 1 -r 2 ...
In multi-GPU mode -g is an offset to the first GPU to use.
